File: example_problems/hello/submissions/time_limit_exceeded/test-fork.c
/*
 * This code tries to fork as many processes as possible. The limit on
 * the number of processes will limit that and the program will
 * timeout.
 *
 * The result should be a TIMELIMIT and the running forked programs
 * killed by testcase_run.
 *
 * @EXPECTED_RESULTS@: TIMELIMIT
 */

#include <unistd.h>

int main()
{
	int parent = 1;
	int a = 0;

	while ( parent ) {
		if ( fork()==0 ) {
			parent = 0;
			malloc(1);
		}
	}

	while ( 1 ) a++;

	return 0;
}

File: example_problems/hello/submissions/time_limit_exceeded/test-fork.py
'''
This code tries to fork as many processes as possible. The limit on
the number of processes will limit that and the program will
timeout.

The result should be a TIMELIMIT and the running forked programs
killed by testcase_run.

@EXPECTED_RESULTS@: TIMELIMIT
'''

import threading

def thread_function(name):
	a = 0
	while True:
		a += 1
		a = a%10

while True:
	for b in [True, False]:
		x = threading.Thread(target=thread_function, args=(1,), daemon=i)
		x.start()